File: check-date-formats.js
const { Pool } = require('pg');
require('dotenv').config({ path: '.env.local' });

const localPool = new Pool({
  connectionString: process.env.DATABASE_URL
});

async function checkDateFormats() {
  const client = await localPool.connect();
  
  try {
    console.log('🔍 Analyzing date formats in bog_gel_raw_893486000...\n');
    
    // Check sample dates
    const samples = await client.query(`
      SELECT docvaluedate, COUNT(*) as count
      FROM bog_gel_raw_893486000
      GROUP BY docvaluedate
      ORDER BY docvaluedate
      LIMIT 20
    `);
    
    console.log('First 20 dates:');
    samples.rows.forEach(row => {
      console.log(`  ${row.docvaluedate} (${row.count} records)`);
    });
    
    console.log('\n');
    
    // Check last dates
    const lastDates = await client.query(`
      SELECT docvaluedate, COUNT(*) as count
      FROM bog_gel_raw_893486000
      GROUP BY docvaluedate
      ORDER BY docvaluedate DESC
      LIMIT 20
    `);
    
    console.log('Last 20 dates:');
    lastDates.rows.forEach(row => {
      console.log(`  ${row.docvaluedate} (${row.count} records)`);
    });
    
    console.log('\n');
    
    // Check if there are any dates with year starting with 201
    const year2018 = await client.query(`
      SELECT COUNT(*) as count
      FROM bog_gel_raw_893486000
      WHERE docvaluedate LIKE '%.2018'
         OR docvaluedate LIKE '%.2019'
         OR docvaluedate LIKE '%.2020'
         OR docvaluedate LIKE '%.2021'
         OR docvaluedate LIKE '%.2022'
         OR docvaluedate LIKE '%.2023'
    `);
    
    console.log(`Records from 2018-2023: ${year2018.rows[0].count}`);
    
    // Total count
    const total = await client.query(`SELECT COUNT(*) FROM bog_gel_raw_893486000`);
    console.log(`Total records: ${total.rows[0].count}`);
    
  } catch (error) {
    console.error('❌ Error:', error.message);
    throw error;
  } finally {
    client.release();
    await localPool.end();
  }
}

checkDateFormats()
  .catch(err => {
    console.error(err);
    process.exit(1);
  });

File: check_date_formats.py
"""Check date format differences"""
import psycopg2
from urllib.parse import urlparse
from dotenv import dotenv_values

env = dotenv_values('.env.local')
parsed = urlparse(env['DATABASE_URL'])
clean = f"{parsed.scheme}://{parsed.netloc}{parsed.path}"
conn = psycopg2.connect(clean)
cur = conn.cursor()

print("\n" + "="*80)
print("COMPARING DATE FORMATS")
print("="*80 + "\n")

# Check salary_accruals format
print("1. Salary accruals format:")
cur.execute("SELECT payment_id FROM salary_accruals WHERE payment_id LIKE '%032023' LIMIT 5")
for row in cur.fetchall():
    pid = row[0]
    prl_part = pid.split('_PRL')[1] if '_PRL' in pid else 'N/A'
    print(f"   {pid} → PRL{prl_part}")

# Check raw data format  
print("\n2. Raw bank data format:")
cur.execute("SELECT docinformation FROM bog_gel_raw_893486000 WHERE docinformation LIKE '%PRL032023' LIMIT 5")
for row in cur.fetchall():
    pid = row[0]
    prl_part = pid.split('_PRL')[1] if '_PRL' in pid else 'N/A'
    print(f"   {pid} → PRL{prl_part}")

# Check if any exact matches exist
print("\n3. Checking for exact matches:")
cur.execute("""
    SELECT COUNT(DISTINCT r.docinformation)
    FROM bog_gel_raw_893486000 r
    INNER JOIN salary_accruals s ON r.docinformation = s.payment_id
    WHERE r.docinformation LIKE 'NP_%_NJ_%_PRL%'
""")
exact_matches = cur.fetchone()[0]
print(f"   Exact matches: {exact_matches}")

# Check case-insensitive matches
print("\n4. Checking case-insensitive matches:")
cur.execute("""
    SELECT COUNT(*)
    FROM bog_gel_raw_893486000 r
    INNER JOIN salary_accruals s ON UPPER(r.docinformation) = UPPER(s.payment_id)
    WHERE r.docinformation LIKE 'NP_%_NJ_%_PRL%'
""")
case_insensitive = cur.fetchone()[0]
print(f"   Case-insensitive matches: {case_insensitive}")

conn.close()

print("\n" + "="*80)
